[Melissa Anastasia Harijanto] iP

melissaharijanto opened the pull request:
# Duke: Task Manager
> "Let our advance worrying become advance thinking and planning." — _[Winston Churchill](https://www.britannica.com/biography/Winston-Churchill)_

Having trouble remembering all of your tasks? Don't worry, __Duke: Task Manager__ is here to _help_ you! 

## Features
With Duke: Task Manager, you can:
- _add_ three types of tasks
- _mark_ them as done, or undo them by _unmarking_
- _list_ all of your tasks
- _find_ your tasks using a keyword
- _delete_ your tasks

All with a _**single command**_!

## Installation Guide 
Here is a guide to run Duke:
1. Download the `.jar` file [here](https://github.com/melissaharijanto/ip/releases/tag/A-Jar).
2. Double-click the file.
3. You are done! 👍 

## Notes
If you are an aspiring Java Programmer taking baby steps to mastering OOP, this project might be able to help you!
A snippet of the Duke class:
```ruby
public Duke() {
    this.tasks = new TaskList();
    this.storage = new Storage();
    this.ui = new Ui();
}

public static void main(String[] args) {
    Duke duke = new Duke();
    duke.run();
}
```

## Your Checklist! ✅ 
- [ ] Install Duke
- [ ] Manage your tasks with Duke
- [ ] Be better at managing your work!

Commit:
Add Gradle support

melissaharijanto committed:
Level-1

melissaharijanto committed:
Level-1 (Duke was not previously commited)

melissaharijanto committed:
Level-2: added Task class and implementation to chat bot

melissaharijanto committed:
Level-2: changed implementation to use ArrayList instead of String[]

melissaharijanto committed:
Level 3: Mark tasks implementation

melissaharijanto committed:
Level 3: Unmark tasks implementation

melissaharijanto committed:
Level-4 & A-Inheritance Tasks and Inheritance

melissaharijanto committed:
Level-4: Implemented error handling for all commands.

melissaharijanto committed:
Level-6: Delete

melissaharijanto committed:
A-TextUiTesting

melissaharijanto committed:
More Error Handling

melissaharijanto committed:
A-Enums: changed if cases to switch cases

melissaharijanto committed:
Made a few changes to code

Description:
- multiple try and catch blocks are condensed into one;
- the command BYE is added to the switch case

melissaharijanto committed:
Catered to CS2103T Java Coding Standards

Description:
- Changed some variables to have names that suit the coding standards
- Added Javadoc according to the standards in the website

melissaharijanto committed:
Fixed indentations of the switch case in Duke

melissaharijanto committed:
Fixes bugs

Description:
- Switch case 'MARK' does not have a colon; added it to fix bug

melissaharijanto committed:
Level-7: Add read, write, save functionality

melissaharijanto committed:
Level-8: Add dates and times

melissaharijanto committed:
Level-8: Delete unused imports and methods

melissaharijanto committed:
Resolve merge conflicts (Level-8 Branch)

melissaharijanto committed:
Add more classes for OOP

melissaharijanto committed:
Divide the classes into packages

melissaharijanto committed:
Merge commit '556af3f47a96b32898ab4cdbd65b16486a4871e8'

* commit '556af3f47a96b32898ab4cdbd65b16486a4871e8':
  Add Gradle support

melissaharijanto committed:
Add gradle and checkstyle

melissaharijanto committed:
Add JUnit Test Cases

melissaharijanto committed:
Add JavaDocs to code

melissaharijanto committed:
Fix the code according to coding standards

melissaharijanto committed:
Implement the find command

melissaharijanto committed:
Fix merge conflicts (branch: A-CodingStandard)

melissaharijanto committed:
Resolve merge conflicts (branch: branch-Level-9)

melissaharijanto committed:
Implement checkstyle and fix code

melissaharijanto committed:
Add JavaFX setup

melissaharijanto committed:
Make changes to code.

Description:
- Change the bot's name from 'Duke' to 'Isara'
- Get the bot to start replying correctly
- Implement JavaFX

melissaharijanto committed:
Improve UI of Isara

melissaharijanto committed:
Add JavaDoc & exit GUI upon bye command

jwdavis0200 commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,147 @@
+     *
+     * @param tasks The list of tasks to be printed.
+     */
+    public void list(TaskList tasks) {
```
Hmm, I would say this is fine but maybe we can be more explicit with the naming(I know, really knitty gritty, would say
this is fine honestly), say maybe name it listOut(..)?

jwdavis0200 commented on src/main/java/duke/command/Command.java:
```diff
@@ -0,0 +1,19 @@
+package duke.command;
```
I see you have done the optional tasks, nice! This is impressive.

jwdavis0200 commented on src/main/java/duke/processor/Parser.java:
```diff
@@ -0,0 +1,149 @@
+ */
+public class Parser {
+    /** Enum class for the available command lines. */
+    enum CommandLine {
```
Great job on using enums!

jwdavis0200 commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,68 @@
+    /**
+     * Unmarks a task as undone.
+     */
+    public void unmarkAsDone() {
```
hmm, I would consider just using mark and unmark as the method names? unmark as done might be slightly confusing?
but again, really digging around to comment on here, so no worries.

jwdavis0200 commented on src/test/java/duke/ToDoTest.java:
```diff
@@ -0,0 +1,38 @@
+        assertEquals(toDo.toString(), "[T][ ] Run in the Morning");
+        assertEquals(toDo.getStatusIcon(), " ");
+    }
+}
```
I really like how you're extensively testing your program, this is definitely good practice for future developments!

jwdavis0200 reviewed (commented):
Hi, You're my first PR review so it's just mostly checking chreally LGTM!! Really impressed with the organisation and the meeting of most if not all checkstyles. Impressive work doing most of the optional tasks as well. Good luck for the rest of the module, am sure you will have no problem.

melissaharijanto committed:
Add assertions

melissaharijanto committed:
Fix code quality

There are several methods that take more than 30 lines. This makes the
makes the code less readable.

There are also some feedback from the code reviewer from the previous
iteration of the iP.

Let's,
* split code in the if block to a different function, if necessary
* change a few of the method names to avoid confusion
* reword parts of the JavaDoc to ensure better documentation for the reader

melissaharijanto committed:
Merge pull request #1 from melissaharijanto/branch-A-Assertions

Add assertions

melissaharijanto committed:
Merge pull request #2 from melissaharijanto/branch-A-CodeQuality

Fix code quality

melissaharijanto committed:
Add CI

melissaharijanto committed:
Fix errors from Checkstyle

melissaharijanto committed:
Rearrange imports for test files

melissaharijanto committed:
Rearrange imports for test files

melissaharijanto committed:
Update yml file

melissaharijanto committed:
Remove I/O from YML file

melissaharijanto committed:
Update icons for the chat bot

melissaharijanto committed:
Remove unused line field from Ui class

melissaharijanto committed:
Add reschedule command / feature

melissaharijanto committed:
Merge pull request #3 from melissaharijanto/branch-B-Snooze

branch-B-Snooze

melissaharijanto committed:
Update README

melissaharijanto committed:
Merge branch 'master' of github.com:melissaharijanto/ip

melissaharijanto committed:
Edit GUI to be able to wrap text

Previously, the GUI was not able to wrap around long texts.

Editing the GUI so that they can wrap around long texts can provide
better user experience, especially when they are listing their
tasks.

As a step towards better user experience, let's edit the FXML file in order
to assist user in listing their many tasks.

melissaharijanto committed:
Add consistent personality to error messages

melissaharijanto committed:
Add code credit (for CompletableFuture in MainWindow)

melissaharijanto committed:
Update user guide

melissaharijanto committed:
Fix align of Ui.png in doc/README

melissaharijanto committed:
Remove horizontal lines and stray <hr> tags from README

melissaharijanto committed:
Fix broken links in README.md

melissaharijanto committed:
Fix more broken links in README.md

melissaharijanto committed:
Remove question mark emoji from README

melissaharijanto committed:
Fix expected output for user guide

melissaharijanto committed:
Generate .jar file

melissaharijanto committed:
Change .jar file version in build.gradle and generate a new file

melissaharijanto committed:
Update user guide to contain the link to the release

melissaharijanto committed:
Fix issues raised by the code quality analysis

melissaharijanto committed:
Fix formatting of user guide
